File: app/src/main/java/java/android/notes/wrapper/fragments/core/NoteFragment.java
package java.android.notes.wrapper.fragments.core;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import java.android.notes.R;
import java.android.notes.activity.IDatatSourseHandler;
import java.android.notes.saveout.IPreferences;
import java.android.notes.core.Control;
import java.android.notes.core.Note;
import java.android.notes.saveout.IWebStore;
import java.android.notes.wrapper.helpers.CreateFragment;
import java.android.notes.wrapper.helpers.Extra;

public class NoteFragment extends Fragment implements View.OnClickListener{
    private Control control;

    EditText editTextHeadLine,editTextDescription,editTextBody;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        setHasOptionsMenu(true);

        return inflater.inflate(R.layout.fragment_note,container,false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        control = ((IDatatSourseHandler)getActivity()).getControl();

        Extra.initToolbar((AppCompatActivity) getActivity(),R.id.toolbarNote);

        editTextHeadLine = view.findViewById(R.id.editTextHeadLine);
        editTextDescription = view.findViewById(R.id.editTextDescription);
        editTextBody = view.findViewById(R.id.editTextBody);

        Note note = control.note;
        editTextHeadLine.setText(note.getHeadline());
        editTextDescription.setText(note.getDescription());
        editTextBody.setText(note.getBody());

        view.findViewById(R.id.buttonSave).setOnClickListener(this);
        view.findViewById(R.id.buttonCancel).setOnClickListener(this);
    }

    @Override
    public void onCreateOptionsMenu(@NonNull Menu menu, @NonNull MenuInflater inflater) {
        super.onCreateOptionsMenu(menu, inflater);

        inflater.inflate(R.menu.menu_fragment_note,menu);
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()){
            case R.id.buttonSave:
                boolean save = control.save(
                    editTextHeadLine.getText().toString(),
                    editTextDescription.getText().toString(),
                    editTextBody.getText().toString()
                );

                if(save){
                    //((IPreferences)requireActivity()).putStringControl();  // save out
                    ((IWebStore)getActivity()).getWebStore().add(control.note); // save out

                    requireActivity().getSupportFragmentManager().popBackStack();
                    requireActivity().getSupportFragmentManager().popBackStack();
                    CreateFragment.createNotesFragment( (AppCompatActivity)requireActivity() );

                }else{
                    Toast.makeText(getContext(), "FILL NOTES'NAME", Toast.LENGTH_SHORT).show();
                }
                break;

            case R.id.buttonCancel:
                requireActivity().getSupportFragmentManager().popBackStack();
                break;
        }

    }


}

File: app/src/main/java/java/android/notes/wrapper/fragments/core/NoteFragment.kt
package java.android.notes.wrapper.fragments.core

import android.widget.EditText
import android.os.Bundle
import android.view.*
import java.android.notes.R
import java.android.notes.activity.IDatatSourseHandler
import java.android.notes.wrapper.helpers.Extra
import androidx.appcompat.app.AppCompatActivity
import java.android.notes.saveout.IPreferences
import android.widget.Toast
import androidx.fragment.app.Fragment
import java.android.notes.core.Control

class NoteFragment : Fragment(), View.OnClickListener {
    private var control: Control? = null
    private var editTextHeadLine: EditText? = null
    private var editTextDescription: EditText? = null
    private var editTextBody: EditText? = null
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        setHasOptionsMenu(true)
        return inflater.inflate(R.layout.fragment_note, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        control = (activity as IDatatSourseHandler).getterControl()
        Extra.initToolbar(activity as AppCompatActivity, R.id.toolbarNote)
        editTextHeadLine = view.findViewById<EditText>(R.id.editTextHeadLine)
        editTextDescription = view.findViewById<EditText>(R.id.editTextDescription)
        editTextBody = view.findViewById<EditText>(R.id.editTextBody)
        val note = control!!.note

        editTextHeadLine?.setText(note.headline)
        editTextDescription?.setText(note.description)
        editTextBody?.setText(note.body)
        view.findViewById<View>(R.id.buttonSave).setOnClickListener(this)
        view.findViewById<View>(R.id.buttonCancel).setOnClickListener(this)
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        super.onCreateOptionsMenu(menu, inflater)
        inflater.inflate(R.menu.menu_fragment_note, menu)
    }

    override fun onClick(view: View) {
        when (view.id) {
            R.id.buttonSave -> {
                val save = control!!.save(
                    editTextHeadLine!!.text.toString(),
                    editTextDescription!!.text.toString(),
                    editTextBody!!.text.toString()
                )
                if (save) {
                    //Toast.makeText(context,"dsds", Toast.LENGTH_SHORT).show()
                    //(requireActivity() as IPreferences).putStringControl() // save out
                    requireActivity().supportFragmentManager.popBackStack()
                } else {
                    Toast.makeText(context, "FILL NOTES'NAME", Toast.LENGTH_SHORT).show()
                }
            }
            R.id.buttonCancel -> requireActivity().supportFragmentManager.popBackStack()
        }
    }
}